clippy_lints/src/invalid_build_cfg.rs

```diff
@@ -0,0 +1,308 @@
+use rustc_ast::ast::{Attribute, MacCall};
+use rustc_ast::token::{LitKind, TokenKind};
+use rustc_ast::tokenstream::{TokenStream, TokenTree};
+use rustc_ast::MetaItem;
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass, LintContext};
+use rustc_session::declare_lint_pass;
+use rustc_span::{sym, Span, Symbol};
+
+use clippy_utils::diagnostics::{span_lint, span_lint_and_sugg};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of `#[cfg]`/`#[cfg_attr]`/`cfg!()` in `build.rs` scripts.
+    ///
+    /// ### Why is this bad?
+    /// It checks the `cfg` values for the *host*, not the target. For example, `cfg!(windows)` is
+    /// true when compiling on Windows, so it will give the wrong answer if you are cross compiling.
+    /// This is because build scripts run on the machine performing compilation, rather than on the
+    /// target.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// if cfg!(windows) {}
+    /// ```
+    /// Use instead:
+    /// ```no_run
+    /// if std::env::var("CARGO_CFG_WINDOWS").is_ok() {}
+    /// ```
+    #[clippy::version = "1.80.0"]
+    pub INVALID_BUILD_CFG,
```

Comment:
Given that there are legitimate use cases for host-config in `build.rs`, I think it would be better if the lint name acknowledges it. the linked issue proposed `misleading_cfg_in_build_script`.
```suggestion
    pub MISLEADING_CFG_IN_BUILD_SCRIPT,
```